### PR TITLE
refactor: cleanup `InstanceError`

### PR DESCRIPTION
**Pull request by magnus-madsen**
- move some tips into the actual error message
- add more colors
- reword a few things
- sort

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala [hunk 1]
```diff
@@ -137,53 +110,53 @@ object InstanceError {
...
       import formatter._
       s"""${line(kind, source.name)}
-         |>> Duplicate type variable '${red(FormatType.formatType(tvar))}' in '${red(sym.name)}'.
          |
-         |${code(loc, s"The type variable '${FormatType.formatType(tvar)}' occurs more than once.")}
+         |>> Illegal use of associated type '${red(assoc.name)}' in instance declaration for '${magenta(clazz.name)}'.
+         |
+         |${code(loc, s"illegal use of associated type")}
+         |
+         |A type class instance cannot use an associated type. Use the full type.
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:130]: "full type" is strange wording here.

File: main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala [hunk 2]
```diff
@@ -292,105 +271,136 @@ object InstanceError {
...
-         |The constraint ${FormatTypeConstraint.formatTypeConstraint(tconstr)} is required because it is a constraint on super class ${superClass.name}.
+         |>> Unlawful signature '${red(sym.name)}'.
          |
-          |${code(loc, s"missing type constraint")}
-      """.stripMargin
+         |>> Each signature of a lawful class must appear in at least one law.
+         |
+         |${code(loc, s"unlawful signature")}
+         |
+         |Create a law for '$sym' or mark the class as unlawful.
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:375]: "unlawful" isn't a keyword anymore

**Review by mlutze — CHANGES_REQUESTED** [review]

File: main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:130]: Not my wording :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:130]: Suggestion?

File: main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala]: Fixed

File: main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala (hunk 1 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:130]: Maybe just remove the last sentence.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala:130]: :D

**Comment by magnus-madsen** [comment]
Can you approve?

**Review by mlutze — APPROVED** [review]

**Pull request merged**